core/src/env_platform/db_handler.bal

```diff
@@ -284,12 +384,29 @@ function saveApplicationMetadata(string applicationType) returns boolean|error {
     }
 }
 
+# The `removeApplicationMetadata` function will remove application metadata count by one from the database.
+# 
+# + applicationType - Type of the application.
+# + return - This function will return either whether the application meta data is removed or 
+# error if there is a mongodb:DatabaseError.
+function removeApplicationMetadata(string applicationType) returns boolean|error {
```

Comment:
If this function only subtracts the count value, I think this function name could be improved to something like **updateApplicationCount**

Reply:
Then will add another param to indicate the count and the operation whether to subtract or add. @deshankoswatte What do you think?

Reply:
@sanjulamadurapperuma what about your thoughts?

Reply:
If we are adding the count, it will be to add Application metadata right? Any other cases where an addition is needed?

Reply:
Also, will there be cases where more than one application metadata is being deleted at a given time?

Reply:
Yeah then will go with the updateApplicationCount

Reply:
If functionality to add and remove metadata are to be put in the same method then +1 to change to updateApplicationCount, but I don't see why we need to pass in the count as a parameter given the current flow of the code. Isn't it always 1?

Reply:
mm yes. Coz there are no situation to add multiple applications or remove at the same time

Reply:
will update with the newer PR